File: pyiArduinoI2Cencoder/cpp/KeyTrigger.cpp
/*
 *	ПРИМЕР РАБОТЫ С КНОПОЧНЫМ ПЕРЕКЛЮЧАТЕЛЕМ:
 * При каждом нажатии на кнопку булева переменная
 * будет менять своё значение на противоположное
 */
#include <iostream>
// Подключаем библиотеку для работы с энкодером I2C-flash.
#include "../iarduino_I2C_Encoder.h"
// Инстанциируем объект, указывая адрес модуля на шине I2C.
iarduino_I2C_Encoder enc(0x09);

void loop(void);

int main()
{
	enc.begin();
	for (;;)
		loop();
}
void loop()
{
	//   Считываем состояние кнопочного переключателя энкодера
	bool f = !!enc.getButton(KEY_TRIGGER);
	std::cout << f << '\n';
	delay(100);
}

File: pyiArduinoI2Cencoder/examples/KeyTrigger.py
#   ПРИМЕР РАБОТЫ С КНОПОЧНЫМ ПЕРЕКЛЮЧАТЕЛЕМ:
# При каждом нажатии на кнопку булева переменная
# будет менять своё значение на противоположное

from time import sleep
# Подключаем библиотеку для работы с энкодером I2C-flash.
from pyiArduinoI2Cencoder import *
# Инстанцируем объект, указывая адрес модуля на шине I2C.
enc = pyiArduinoI2Cencoder(0x09)

while True:
    # Считываем состояние кнопочного переключателя энкодера
    t = bool(enc.getButton(KEY_TRIGGER))
    print(t)
    sleep(.1)
